cosmwasm/contracts/global-accountant/src/contract.rs

```diff
@@ -334,9 +334,21 @@ fn handle_vaa(
     let mut evt = if body.emitter_chain == Chain::Solana
         && body.emitter_address == wormhole::GOVERNANCE_EMITTER
     {
-        let govpacket = serde_wormhole::from_slice(body.payload)
-            .context("failed to parse governance packet")?;
-        handle_governance_vaa(deps.branch(), info, body.with_payload(govpacket))?
+        let mut module = [0u8; 32];
+        module.clone_from_slice(&body.payload.get()[..32]);
+
+        if let Ok(govpacket) = serde_wormhole::from_slice::<token::GovernancePacket>(body.payload) {
```

Comment:
This might be better written as `match` where the default (`_`) match results in `bail!`.

`if let` is a nice syntax when there aren't other conditions (either match something specific, otherwise do something else).

Reply:
match won't work here since we need to actually call `serde_wormhole::from_slice:` twice.  Once for `token::GovernancePacket` and once for `accountant::GovernancePacket`.